instaclustr/resource_cluster_unit_test.go

```diff
@@ -0,0 +1,99 @@
+package instaclustr
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+)
+
+func TestCreateBundleUserUpdateRequest(t *testing.T) {
+	testUsername := "hail"
+	testPassword := "reallySecure123"
+	testBundleRequest := createBundleUserUpdateRequest(testUsername, testPassword)
+
+	expectedOutput := fmt.Sprintf("{\"username\":\"%s\",\"password\":\"%s\"}", testUsername, testPassword)
+
+	if fmt.Sprintf("%s", testBundleRequest) != expectedOutput {
+		t.Fatalf("Incorrect request returned.\nExpected:%s\nActual:%s", expectedOutput, testBundleRequest)
+	}
```

Comment:
**[Desired, Maintainability]** It doesn't feel right that we are comparing manually stringified JSON. I would prefer to see it done on a bytes marshalled basis:
```go
func JSONBytesEqual(a, b []byte) (bool, error) {
    var j, j2 interface{}
    if err := json.Unmarshal(a, &j); err != nil {
        return false, err
    }
    if err := json.Unmarshal(b, &j2); err != nil {
        return false, err
    }
    return reflect.DeepEqual(j2, j), nil
}

func main() {
    b1 := []byte(`{"x": {"t": 1, "s": 2}, "z": 1}`)
    b2 := []byte(`{"z": 1, "x": {"s": 2, "t": 1}}`)

    eq, err := JSONBytesEqual(b1, b2) // false
    eq, err = JSONBytesEqual(b1, b1) // true
}
```

Reply:
addressed - comparing the bytes slice so we dont need any data processing for test